docs/usage/services.rst

```diff
@@ -0,0 +1,169 @@
+========
+Services
+========
+
+Services in Advanced Alchemy build upon repositories to provide higher-level business logic, data transformation,
+and schema validation. While repositories handle raw database operations, services handle the application's
+business rules and data transformation needs.
+
+Understanding Services
+----------------------
+
+Services provide:
+
+- Business logic abstraction
+- Data transformation using Pydantic or Msgspec models
+- Input validation
+- Complex operations involving multiple repositories
+- Consistent error handling
+
+Basic Service Usage
+-------------------
+
+Let's enhance our blog example with services:
+
+.. code-block:: python
+
+    from advanced_alchemy.service import SQLAlchemyAsyncRepositoryService
+    from sqlalchemy.ext.asyncio import AsyncSession
+    from pydantic import BaseModel
+    from typing import Optional
+    from uuid import UUID
+
+    # Define DTOs for input/output
+    class PostCreate(BaseModel):
+        title: str
+        content: str
+        author_id: UUID
+        published: bool = False
+
+    class PostRead(BaseModel):
+        id: int
+        title: str
+        content: str
+        author_id: UUID
+        published: bool
+        created_at: datetime
+        updated_at: datetime
+
+    class PostService(SQLAlchemyAsyncRepositoryService[Post]):
+        """Service for managing blog posts with data transformation."""
+
+        model_type = Post
+
+        # Define schema mappings
+        create_schema = PostCreate
+        read_schema = PostRead
+
+    async def create_post(session: AsyncSession, data: PostCreate) -> PostRead:
+        service = PostService(session=session)
+        # Service automatically handles validation and transformation
+        return await service.create(data)
+
+Data Transformation
+-------------------
+
+Services automatically handle transformation between DTOs and database models:
+
+.. code-block:: python
+
+    from msgspec import Struct
+    from typing import List
+
+    # Using msgspec for better performance
+    class PostUpdate(Struct):
+        title: Optional[str] = None
```

Comment:
Do we do the Python type versioning here so it'll have tabs for `str | None`, etc. like we do in Litestar?

Reply:
yep. we do.  I noticed this in my review earlier.